finetune/adapter.py

```diff
@@ -47,18 +48,34 @@
 }
 
 
-def main(
+def setup(
     data_dir: Path = Path("data/alpaca"),
     checkpoint_dir: Path = Path("checkpoints/stabilityai/stablelm-base-alpha-3b"),
     out_dir: Path = Path("out/adapter/alpaca"),
     precision: Literal["bf16-true", "32-true"] = "bf16-true",
+    tpu: bool = False
 ):
-    check_valid_checkpoint_dir(checkpoint_dir)
-
+    strategy = (DeepSpeedStrategy(config=ds_config) if (devices > 1 and not tpu) else "auto")
     fabric = L.Fabric(
-        devices=devices, strategy=(DeepSpeedStrategy(config=ds_config) if devices > 1 else "auto"), precision=precision
+        devices=devices, strategy=strategy, precision=precision
     )
-    fabric.launch()
+    
+    if tpu:
+        p_main = functools.partial(main, data_dir, checkpoint_dir, out_dir)
+        fabric._strategy._sync_module_states = False
```

Comment:
Why not instantiate with `XLAStrategy(sync_module_states=False)` if `tpu is True`?

Reply:
This makes more sense, I will add a commit fixing this.